plugins/volume/org.xen.xapi.storage.zfs-ng/sr.py

```diff
@@ -0,0 +1,226 @@
+#!/usr/bin/env python
+
+import os
+import os.path
+import sys
+import urlparse
+
+from xapi.storage import log
+from xapi.storage.libs import util
+from xapi.storage.libs.libcow.zfsutil import ZFSUtil
+from xapi.storage.common import call
+from xapi.storage.libs.libcow.imageformat import ImageFormat
+from xapi.storage.libs.libcow.volume import COWVolume
+from xapi.storage.libs.libcow.callbacks import VolumeContext
+import xapi.storage.api.v5.volume
+
+import importlib
+
+ZPOOL_BIN = 'zpool'
+ZFS_BIN = 'zfs'
+
+@util.decorate_all_routines(util.log_exceptions_in_function)
+class Implementation(xapi.storage.api.v5.volume.SR_skeleton):
+    def probe(self, dbg, configuration):
+        return {
+            'srs': [],
+            'uris': []
+        }
+
+    def attach(self, dbg, configuration):
+        uri = configuration['mountpoint']
+        log.debug('{}: SR.attach: config={}, uri={}'.format(
+            dbg, configuration, uri))
+
+        sr = urlparse.urlparse(uri).path
+        return sr
+
+    def create(self, dbg, sr_uuid, configuration, name, description):
+        log.debug('{}: SR.create: config={}, sr_uuid={}'.format(
+            dbg, configuration, sr_uuid))
+
+        if 'devices' not in configuration:
+            log.error('devices parameter is missed')
+            raise
+        devs = configuration['devices'].split(',')
+
+        mountpoint = '/' + name
+        if 'mountpoint' in configuration:
+            mountpoint = configuration['mountpoint']
+
+        compression = False
+        if 'compression' in configuration:
+            if configuration['compression'] in [
+                    'true', 't', 'on', '1', 'yes']:
+                compression = True
+
+        cmd = [
+            ZPOOL_BIN, 'create', '-f',
+            name, '-m', mountpoint
```

Comment:
You can use a method defined in ZFSUtil. :)

Reply:
ZFSUtil.py:create() uses ZFS_BIN instead of ZPOOL_BIN. That's why I do not use ZFSUtil for ZPOOL_BIN commands.

Reply:
I implemented a method named create_pool() in ZFSUtil to do this.